torchmetrics/functional/regression/mean_absolute_percentage_error.py

```diff
@@ -41,11 +40,34 @@ def _mean_absolute_percentage_error_compute(sum_abs_per_error: Tensor, num_obs:
     return sum_abs_per_error / num_obs
 
 
-def mean_absolute_percentage_error(preds: torch.Tensor, target: torch.Tensor, eps: float= 1.17e-07) -> Tensor:
-    """something"""
+def mean_absolute_percentage_error(preds: torch.Tensor, target: torch.Tensor, epsilon: float = 1.17e-07) -> Tensor:
+    """
+    Computes mean absolute percentage error.
+
+    Args:
+        preds: estimated labels
+        target: ground truth labels
+        epsilon: an arbitrary small yet strictly positive number to avoid undefined results when y is zero.
+                 default: 1.17e-07. For more information, check the note.
+
+    Return:
+        Tensor with MAPE
+
+    Note:
+        The epsilon value is taken from `scikit-learn's
+        implementation
+        <https://github.com/scikit-learn/scikit-learn/blob/15a949460/sklearn/metrics/_regression.py#L197>`_.
+
+    Example:
+        >>> from torchmetrics.functional import mean_absolute_percentage_error
+        >>> target = torch.tensor([1, 10, 1e6])
+        >>> preds = torch.tensor([0.9, 15, 1.2e6])
+        >>> mean_absolute_percentage_error(preds, target)
+        tensor(0.2667)
+    """
 
-    eps = torch.tensor(eps)
-    sum_abs_per_error, num_obs = _mean_absolute_percentage_error_update(preds, target, eps)
+    epsilon = torch.tensor(epsilon)
```

Comment:
does it need to be a tensor? Seems like maybe fine to have a regular float?

Reply:
So, I had initially used a normal float, but it would result in this error ```TypeError: max() received an invalid combination of arguments - got (float, Tensor)```. Hence the conversion to tensor!

Reply:
OK, then you should at least do `tensor(epsilon, device=preds.device)` otherwise you'll get an error about tensors being on different devices once you start running this on GPU.

Reply:
after looking at the code again, it does not make sense to have `epsilon` as an argument. It is not something that the user should be able to control. It just to secure we do not divide by 0 :]